Userland/Libraries/LibGfx/TGALoader.cpp

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2022, Tom Needham <06needhamt@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Span.h>
+#include <AK/String.h>
+#include <LibGfx/TGALoader.h>
+
+namespace Gfx {
+
+enum TGADataType : u8 {
+    None = 0,
+    UncompressedColorMapped = 1,
+    UncompressedRGB = 2,
+    UncompressedBlackAndWhite = 3,
+    RunLengthEncodedColorMapped = 9,
+    RunLengthEncodedRGB = 10,
+    CompressedBlackAndWhite = 11,
+    CompressedColorMapped = 32,
+    CompressedColorMappedFourPass = 33
+};
+
+struct TGAHeader {
+    u8 id_length;
+    u8 color_map_type;
+    enum TGADataType data_type_code;
+    i16 color_map_origin;
+    i16 color_map_length;
+    u8 color_map_depth;
+    i16 x_origin;
+    i16 y_origin;
+    u16 width;
+    u16 height;
+    u8 bits_per_pixel;
+    u8 image_descriptor;
+} __attribute__((packed));
+
+static_assert(sizeof(TGAHeader) == 18);
+
+union TGAPixel {
+    struct TGAColor {
+        u8 blue;
+        u8 green;
+        u8 red;
+        u8 alpha;
+    } components;
+
+    u32 data;
+} __attribute__((packed));
+
+static_assert(sizeof(TGAPixel) == 4);
```

Comment:
The Kernel uses `AssertSize<Type, size>`, which has nicer error messages
but this does not need to be changed